File: backend/internal/models/message.go
package models

import "time"

type Conversation struct {
	ID            string    `json:"id"`
	Participant1ID string    `json:"participant1Id"`
	Participant2ID string    `json:"participant2Id"`
	CreatedAt     time.Time `json:"createdAt"`
	UpdatedAt     time.Time `json:"updatedAt"`
}

type Message struct {
	ID             string    `json:"id"`
	ConversationID string    `json:"conversationId"`
	SenderID       string    `json:"senderId"`
	Content        string    `json:"content"`
	Read           bool      `json:"read"`
	CreatedAt      time.Time `json:"createdAt"`
}

File: packages/shared/src/domain/message.ts
/**
 * Message Domain Models
 *
 * TypeScript interfaces matching the Go backend models.
 */

export interface Conversation {
  id: string;
  participant1Id: string;
  participant2Id: string;
  participants?: Array<{ id: string }>;
  createdAt: Date | string;
  updatedAt: Date | string;
}

export interface Message {
  id: string;
  conversationId: string;
  senderId: string;
  content: string;
  read: boolean;
  createdAt: Date | string;
}

export interface CreateConversationRequest {
  userId1: string;
  userId2: string;
}

export interface SendMessageRequest {
  senderId: string;
  content: string;
}

export interface MessageListParams {
  limit?: number;
  offset?: number;
}

export interface ConversationListParams {
  userId: string;
}
